dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentVersionResource.java

```diff
@@ -133,6 +147,69 @@ public Response findVersions(@Context final HttpServletRequest request,
 
     }
 
+    private Identifier getIdentifier(final String identifier, final User user ) throws DotDataException{
+        if(!UtilMethods.isSet(identifier)){
+          return null;
+        }
+        final ShortyId shorty = APILocator
+                .getShortyAPI().getShorty(identifier)
+                .orElseThrow(() -> new DoesNotExistException(
+                        getFormattedMessage(user.getLocale(), FIND_BY_ID_ERROR_MESSAGE_KEY,
+                                identifier)));
+        return
+                (shorty.type == ShortType.IDENTIFIER) ? APILocator.getIdentifierAPI()
+                        .find(shorty.longId)
+                        : APILocator.getIdentifierAPI().findFromInode(shorty.longId);
+    }
+
+    private List<Contentlet> findByInodes(final User user, final Set<String> inodes, final boolean respectFrontendRoles)
+            throws DotStateException {
+
+        final Set<String> notFound = new HashSet<>();
+        final Set<ShortyId> mappedShorties = inodes.stream().map(inode -> {
+            final Optional<ShortyId> shortyOptional = APILocator.getShortyAPI().getShorty(inode);
+            return shortyOptional.orElseGet(() -> {
+                notFound.add(inode);
+                return null;
+            });
+        }).collect(Collectors.toSet());
```

Comment:
might here can filter possible nulls

Reply:
Not really needed, Because if the NotFound Collection contains anything an exception is thrown